app/policies/blog_policy.rb

```diff
@@ -0,0 +1,23 @@
+class BlogPolicy < ApplicationPolicy
+  class Scope < Scope
+    def resolve
+      scope.where(published: true)
+    end
+  end
+
+  def show?
+    true  # Anyone can view blogs
+  end
+
+  def create?
+    true  # Anyone can create blogs (adjust as needed)
```

Comment:
This is a security vulnerability. Any visitor can delete or modify any blog post without authentication.